### PR TITLE
[TACHYON-1668] Tachyon UI is broken because of security checking

### PR DESCRIPTION
**Pull request by dongche**
JIRA: https://tachyon.atlassian.net/browse/TACHYON-1668

Bug fixed. Check whether security is enabled at first. If not, which is the default conf in Tachyon, the if branch should be bypassed.

@apc999 , @yupeng9 , @sundapeng

### CODE REVIEW

File: core/server/src/main/java/tachyon/web/WebInterfaceDownloadServlet.java [hunk 1]
```diff
@@ -69,7 +70,8 @@ public WebInterfaceDownloadServlet(FileSystemMaster fsMaster) {
   @Override
   protected void doGet(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
-    if (PlainSaslServer.AuthorizedClientUser.get() == null) {
+    if (SecurityUtils.isSecurityEnabled(MasterContext.getConf())
```
Comment (yupeng9) [inline core/server/src/main/java/tachyon/web/WebInterfaceDownloadServlet.java]: Would it make sense to check if security is enabled in `PlainSaslServer.AuthorizedClientUser.get()`?

**Comment by yupeng9** [comment]
Thanks for the fix. Are you interested in https://tachyon.atlassian.net/browse/TACHYON-1670, adding a Web Server test to ensure the endpoints not broken?

File: core/server/src/main/java/tachyon/web/WebInterfaceDownloadServlet.java (hunk 1 above)
Comment (dongche) [inline core/server/src/main/java/tachyon/web/WebInterfaceDownloadServlet.java]: Sounds good. Added checking authentication enabled in the method.

For the special handling here, we still have to check if security (authentication + authorization) is enabled explicitly, so as to not impact non-security logic if using default conf.

**Comment by dongche** [comment]
@yupeng9 , thanks for the review. Patch updated.

About the Web Server test, needs some investigation. A quick question is in the tests what we want is to simulate each servlet (corresponds to tabs) instance, send mock request, and verify the response. Right?

File: core/server/src/main/java/tachyon/web/WebInterfaceDownloadServlet.java (hunk 1 above)
Comment (yupeng9) [inline core/server/src/main/java/tachyon/web/WebInterfaceDownloadServlet.java]: Thanks. Yeah, I understand it. Adding the checking is for showing better error message.

**Comment by yupeng9** [comment]
Yes, the web server test is beyond the scope of this PR. What you described is correct: launch the server, mock an HTTP request, and verify the response type.

**Comment by yupeng9** [comment]
LGTM

**Comment by apc999** [comment]
Both two tabs are viewable now---see snapshots attached.

screen shot 2016-01-28 at 11 12 56 am

screen shot 2016-01-28 at 11 12 18 am

**Comment by apc999** [comment]
+1 LGTM!

Thanks for the patch!

**Pull request merged**